backend/src/apiserver/common/sql/dialect/dialect.go

```diff
@@ -0,0 +1,171 @@
+// Copyright 2025 The Kubeflow Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package dialect provides a minimal, shared SQL dialect configuration
+// for apiserver components. It centralizes identifier quoting and
+// placeholder styles for different backends, and returns a Squirrel
+// StatementBuilder configured for the target database.
+package dialect
+
+import (
+	"strings"
+
+	sq "github.com/Masterminds/squirrel"
+)
+
+// DBDialect holds read-only runtime configuration for a SQL backend.
+// All fields are private; callers must use the exported getter methods.
+type DBDialect struct {
+	name                 string
+	quoteIdentifier      func(string) string
```

Comment:
optional: consider making this into a type: 

```
type QuoteFunction func(string) string
```

then also use this in function signatures like `joinQuoted` or `quoteAll`